onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -130,6 +131,32 @@ def optimize_by_onnxruntime(
             ["CUDAExecutionProvider", "ROCMExecutionProvider", "MIGraphXExecutionProvider"]
         )
 
+    # Move original external data file to temp dir if it exists
+    has_external_data_file = False
+    original_model = onnx.load(onnx_model_path, load_external_data=False)
+    for initializer in original_model.graph.initializer:
+        if initializer.HasField("data_location") and initializer.data_location == onnx.TensorProto.EXTERNAL:
+            has_external_data_file = True
+            break
+
+    if has_external_data_file:
+        temp_dir = os.path.dirname(optimized_model_path)
+        original_dir = os.path.dirname(onnx_model_path)
+        original_model_filename = os.path.basename(onnx_model_path)
+        original_model_in_temp_dir_path = os.path.join(temp_dir, original_model_filename)
+
+        onnx.external_data_helper.load_external_data_for_model(original_model, original_dir)
+        onnx.save_model(
+            original_model,
+            original_model_in_temp_dir_path,
+            save_as_external_data=True,
+            all_tensors_to_one_file=True,
+            location=f"{original_model_filename}.data",
+            size_threshold=1024,
```

Comment:
This shall be done before onnxruntime.InferenceSession. The session shall use saved model to create session